core/src/main/java/org/apache/calcite/sql/SqlOperator.java

```diff
@@ -1036,6 +1036,19 @@ public <R> void acceptCall(
     return null;
   }
 
+  /**
+   * Returns whether this is a safe operator that doesn't throw any exception.
+   *
+   * <p> If an operator is a safe operator, we will check its parameters to determine
+   * if the entire expression is safe.
+   *
+   * <p>For example, {@code 1/0 + null} is not safe. Because even though the
+   * {@code SqlStdOperatorTable.PLUS} is safe operator but its operands is not safe.
+   */
+  public Boolean isSafeOperator() {
```

Comment:
the design of this function deserves some discussion

Reply:
This is mainly due to compatibility issues. When a user adds a custom Operator, we default it to no-safe.

Reply:
the comments cannot have "we", there is no "we" when the compiler runs.
this is a very important function, the documentation should be very crisp about what is does.

Reply:
How about this comment:
```
  /**
   * Returns whether this is a safe operator that doesn't throw any exception.
   *
   * <p> If an operator is safe, then the safety of the entire expression
   * will depend on its parameters.
   *
   * <p>For example, the expression {@code 1/0 + null} is not safe. Because even though the
   * {@code SqlStdOperatorTable.PLUS} is safe operator but its parameters is not safe.
   *
   * <p>By default, returns {@code false}, which means the operator is not safe.
   *
   * <p>If an expression is safe, then some optimizations can be performed in
   * {@code org.apache.calcite.rex.RexSimplify}. 
   * 
   * <p>For example:
   * {@code NULL + a} can be optimized to {@code NULL}.
   *
   */
```

Reply:
Yes, I agree with your point. Currently, our design here is independent and not associated with any specific dialect.